components/suggest/src/db.rs

```diff
@@ -651,10 +659,11 @@ impl<'a> SuggestDao<'a> {
         &mut self,
         record_id: &SuggestRecordId,
         suggestions: &[DownloadedAmoSuggestion],
+        ingest_interrupt_scope: &SqlInterruptScope,
```

Comment:
Ahh, I had `SuggestDao` hold an interrupt scope specifically so that we could avoid passing it around 😄

The `Dao` is a transient object that only lives for the lifetime of a `read()` or `write()` call, so I was thinking that, if you wanted a separate interrupt scope for just ingestion, you could create a `SuggestDao` with a separate interrupt scope, and that `SuggestDao` would only live for the lifetime of the ingestion.

WDYT? This isn't blocking by any means; I just wanted to make sure I understood if there were other reasons for passing the scope around explicitly.

Reply:
The transaction gets committed at end of `write` and I didn't want to change that behavior.  But it does seem better to just pass around the `SuggestDao`.  I'll update the PR to create a single DAO for the entire ingestion and add a `SuggestDao::commit` method that we can call.  This seems better all-around, it seems better to see the explicit commit in the code.